Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,463 @@
+            {
+                var currentJob = mind.CurrentJob.Name;
+                currentJob = currentJob.Replace(" ", "");
+                if (mind.OwnedEntity != null && jobs.Roles.Contains(currentJob))
+                {
+                    if (!HasComp<HeadRevolutionaryComponent>(mind.OwnedEntity))
+                    {
+                        EnsureComp<RevolutionaryRuleComponent>(mind.OwnedEntity.Value);
+                        EnsureComp<CommandStaffComponent>(mind.OwnedEntity.Value);
+                    }
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Called when a Head Rev uses a flash in melee to convert somebody else.
+    /// </summary>
+    public void OnPostFlash(EntityUid uid, HeadRevolutionaryComponent comp, ref AfterFlashedEvent ev)
+    {
+        var stunTime = TimeSpan.FromSeconds(3);
+        if (!HasComp<RevolutionaryComponent>(ev.Target) && !HasComp<MindShieldComponent>(ev.Target) && HasComp<HumanoidAppearanceComponent>(ev.Target)
+            && TryComp<MobStateComponent>(ev.Target, out var mobState) && mobState.CurrentState == MobState.Alive)
+        {
+            if (ev.Used != null)
+            {
+                _charges.AddCharges(ev.Used.Value, 1);
+            }
+            _npcFaction.RemoveFaction(ev.Target, "NanoTrasen");
+            _npcFaction.AddFaction(ev.Target, "Revolutionary");
+            EnsureComp<RevolutionaryComponent>(ev.Target);
+            _sharedStun.TryParalyze(ev.Target, stunTime, true);
+            if (ev.User != null)
+            {
+                _adminLogManager.Add(LogType.Mind, LogImpact.Medium, $"{ToPrettyString(ev.User.Value)} converted {ToPrettyString(ev.Target)} into a Revolutionary");
+            }
+
+            var mind = _mindSystem.GetMind(ev.Target);
+            if (mind != null && mind.OwnedEntity != null)
+            {
+                _mindSystem.AddRole(mind, new RevolutionaryRole(mind, _prototypeManager.Index<AntagPrototype>("Rev")));
+                if (mind.Session != null)
+                {
+                    var message = Loc.GetString("rev-role-greeting");
+                    var wrappedMessage = Loc.GetString("chat-manager-server-wrap-message", ("message", message));
+                    _chatManager.ChatMessageToOne(Shared.Chat.ChatChannel.Server, message, wrappedMessage, default, false, mind.Session.ConnectedClient, Color.Red);
+                }
+            }
+        }
+    }
+
+    public void OnHeadRevAdmin(Mind.Mind mind, IPlayerSession headRev)
+    {
+        var revRule = EntityQuery<RevolutionaryRuleComponent>().FirstOrDefault();
+        if (revRule == null)
+        {
+            GameTicker.StartGameRule("Revolutionary");
+        }
+        if (mind.OwnedEntity != null)
+        {
+            EnsureComp<RevolutionaryRuleComponent>(mind.OwnedEntity.Value);
+            GiveHeadRevRole(mind, headRev);
+            AssignCommandStaff();
+        }
+    }
```

Comment:
The verb should be starting the game rule, not here.

Reply:
I tried starting the gamerule in the verb but that wasn't working for me

Reply:
What happens?

Reply:
When I try to use GameTicker.StartGameRule("Revolutionary"); in the verb, it says that an objected reference is required.

Reply:
GameTicker here is a field on GameRuleSystem<T>, youd have to add it as a dependency with `[Dependency] private readonly GameTicker _gameTicker = default`;